include/gz/sim/MeshInertiaCalculator.hh

```diff
@@ -0,0 +1,230 @@
+    }
+
+    for (int i = 0; i < 10; ++i) {
+        integral[i] *= coefficients[i];
+    }
+
+    // Accumulate the result and add it to MassMatrix object of gz::math
+    double volume = integral[0];
+    double mass = volume * density;
+    gz::math::Vector3d center_of_mass = gz::math::Vector3d();
+    center_of_mass.X() = integral[0] / mass;
+    center_of_mass.Y() = integral[1] / mass;
+    center_of_mass.Z() = integral[2] / mass;
+
+    gz::math::Vector3d ixxyyzz = gz::math::Vector3d();
+    gz::math::Vector3d ixyxzyz = gz::math::Vector3d();
+
+    // Diagonal Elements of the Mass Matrix
+    ixxyyzz.X() = integral[5] + integral[6] - mass * (center_of_mass.Y() * center_of_mass.Y() + center_of_mass.Z() * center_of_mass.Z());
+    ixxyyzz.Y() = -(integral[4] + integral[6] - mass * (center_of_mass.Z() * center_of_mass.Z() + center_of_mass.X() * center_of_mass.X()));
+    ixxyyzz.Z() = -(integral[4] + integral[5] - mass * (center_of_mass.X() * center_of_mass.X() + center_of_mass.Y() * center_of_mass.Y()));
+
+    // Off Diagonal Elements of the Mass Matrix
+    ixyxzyz.X() = - (integral[7] - mass * center_of_mass.X() * center_of_mass.Y());
+    ixyxzyz.Y() = - (integral[8] - mass * center_of_mass.Y() * center_of_mass.Z());
+    ixyxzyz.Z() = - (integral[9] - mass * center_of_mass.X() * center_of_mass.Z());
+
+    // Set the values in the MassMatrix object
+    massMatrix.SetMass(mass);
+    massMatrix.SetDiagonalMoments(ixxyyzz);
+    massMatrix.SetOffDiagonalMoments(ixyxzyz);
+    std::cout << "Mass of the mesh is: " << massMatrix.Mass() << std::endl;
+    std::cout << "Diagonal Elements of the Inertia Matrix are: " << massMatrix.DiagonalMoments() << std::endl;
+    std::cout << "Off Diagonal Elements of the Inertia Matrix are: " << massMatrix.OffDiagonalMoments() << std::endl;
+  }
+
+  std::optional<gz::math::MassMatrix3d> MeshInertiaCalculator::operator()(sdf::Errors& _errors,
+          const sdf::InterfaceMoiCalculator& _calculatorParams)
+  {
+    const gz::common::Mesh *mesh = nullptr;
+    const double density = _calculatorParams.Density();
+    const sdf::Mesh sdfMesh = _calculatorParams.Mesh();
+
+    auto fullPath = asFullPath(sdfMesh.Uri(), sdfMesh.FilePath());
+    if (fullPath.empty())
+    {
+      gzerr << "Mesh geometry missing uri" << std::endl;
+      _errors.push_back({sdf::ErrorCode::URI_INVALID,
+          "Attempting to load the mesh but the URI seems to be incorrect"});
+      return std::nullopt;
+    }
+
+    // Load the Mesh
+    gz::common::MeshManager *meshManager = gz::common::MeshManager::Instance();
+    mesh = meshManager->Load(fullPath);
+
+    std::vector<Triangle> meshTriangles;
+    gz::math::MassMatrix3d meshMassMatrix;
+
+    // Create a list of Triangle objects from the mesh vertices and indices
+    GetMeshTriangles(meshTriangles, mesh);
+
+    CalculateMassProperties(meshTriangles, density, meshMassMatrix);
+
+    return meshMassMatrix;
+  }
+}
+}
+}
+
+#endif
+
+
+
```

Comment:
```suggestion
#endif

```

Reply:
Corrected in [ee6f863](https://github.com/gazebosim/gz-sim/pull/2061/commits/ee6f863535a4f9fabd39e914f9bf5861f5d939da)